packages/nitro-protocol/contracts/ninja-nitro/readme.md

```diff
@@ -0,0 +1,97 @@
+# Ninja Nitro
+
+Ninja-Nitro is an alternative implementation of Nitro protocol, with a different contract architecture to the main implementation (Mono-Nitro) in this package. The additional files should all have namespaced paths.
+
+## Overview
+
+### Recap of Mono-Nitro
+
+The main implementation has 1 x monolithic `Adjudicator` contract and m x `AssetHolder` contracts (one for each of the m Assets in the state channel network). The `Adjudicator` API is
+
+- The ForceMove API
+- `pushOutcome`
+- utility methods that just batch up this API and the `AssetHolder` API (`pushOutcomeAndTransferAll` and `concludePushOutcomeAndTransferAll`)
+
+The essence of the architecture, then, is `pushOutcome`: a finalized outcome must be "pushed" from the `Adjudicator`, resulting in a piece of that outcome being stored in each `AssetHolder`. Nevertheless, `pushOutcome` is an implementation detail and not necessary part of the overall Nitro API.
+
+The `AssetHolder` API is:
+
+- `transfer`
+- `claim`
+- permissioned methods `transferAllAdjudicatorOnly` and `setAssetOutcomeHash` only callable from the `Adjudicator`.
+
+After a `pushOutcome` ssets may be released by targetting each `AssetHolder` and calling `claim` or `transfer` (which _are_ essential parts of the overall Nitro API). This is the unhappy path. The happy path will shortcut `pushOutcome` and use the permissioned methods.
+
+### Ninja-Nitro
+
+Ninja-Nitro, by contrast has 1 x `AdjudicatorFactory` contract, 1 x `SingleChannelAdjudicator` master contract, and k x `Proxy` contracts, one for each of the k channels in the network that resolves on chain. The factory and master contracts are "infastructure" and deployed once (most likely by the protocol team).
```

Comment:
`most likely by the protocol team`

This refers to team X who are authoring some N-N based SCDapp?

Reply:
No, I really mean "us". The team that maintains the state channel network. I'll try and make this clearer.

Reply:
24c10f7